jvector-base/src/main/java/io/github/jbellis/jvector/graph/OnHeapGraphIndex.java

```diff
@@ -568,7 +576,8 @@ public static OnHeapGraphIndex load(RandomAccessReader in, int dimension, double
 
         int entryNode = in.readInt();
 
-        var graph = new OnHeapGraphIndex(layerDegrees, dimension, overflowRatio, diversityProvider);
+        var isHierarchical = layerCount > 1;
```

Comment:
This property being exposed as a test of layer count is inconsistent with the idea that a graph may be hierarchical during construction. If it is important for callers to know the intended graph mode during and after construction, then the property should be supplied as a ctor argument and returned instead of this structural test.